### ISSUE TITLE
Relative problem paths don't take `cwd` into account

### ISSUE THREAD

**Issue by WardenGnaw** [Microsoft/vscode#58980]
From https://github.com/Microsoft/vscode-cpptools/issues/2518
by @galgalesh 
--------------------------------------------------------

**Type: LanguageService**

**Describe the bug**
- OS and Version: Ubuntu 18.04
- VS Code Version:  1.27.1 5944e81f3c46a3938a82c701f96d7a59b074cfdc x64
- C/C++ Extension Version: 0.18.1
- Other extensions you installed (and if the issue persists after disabling them):
- A clear and concise description of what the bug is.

The problemMatcher doesn't take the `cwd` of a task into account when resolving relative paths.

**To Reproduce**

Create a task and set the `cwd` of that task to a different path.

**Expected behavior**

The problemMatcher resolves relative paths as relative to `cwd`. If you set the `cwd` of a task, relative paths will most likely be relative to the `cwd`, not to the workspaceFolder.

**Behavior I get**

The problemMatcher resolves relative paths as relative to `workspaceFolder`

Following is the `tasks.json` for such a task. The line starting with ` "fileLocation"` shouldn't be neccisary.

```
        {
            "label": "build adfgxx",
            "type": "shell",
            "command": "ninja",
            "args": [
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "problemMatcher": {
                "base": "$gcc",
                // The next line shouldn't be needed since it is the same value as `cwd`
                "fileLocation" : ["relative", "${workspaceFolder}/builddir"]
            },
            "options": {
                "cwd": "${workspaceFolder}/builddir",
            }
        }
```

**Additional context**

**Comment by dbaeumer** [comment]
I see the point. The problem is that the `cwd` in options is for the process we start and the `cwd` for the problem matcher the current `cwd` of the VS Code process. But I see that in this constellation it would make sense to treat the `cwd` the same. But it might be a breaking change since it will change the behavior for exist tasks if we change this. Need to change this with care.

**Comment by dschaefer** [comment]
Yup, just running into this as well for CDT's vscode extension. I'd be OK with a new variable of compat is a concern.

**Comment by RoyaAoki** [comment]
Got same problem here. Maybe we need another "cwd" for the problemMatcher specifically?
